### PR TITLE
Update FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql

### PR DESCRIPTION
**Pull request by almabekwoldesenbet**
23.2.1 Dec release death load code changes

### CODE REVIEW

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 1]
```diff
@@ -288,9 +567,9 @@ END
 + CASE WHEN BURIAL_PERMIT_NONE = 'Y' THEN ''
 		WHEN BURIAL_PERMIT_NONE = 'N' THEN
 			(CASE WHEN C.CURRENT_CITY_OPT_IN = 'Y' AND BURIAL_PERMIT_ISSUE_DATE IS NULL AND ISDATE(BURIAL_PERMIT_ISSUE_DATE_RV) = 0 
-					THEN REPLACE ((RIGHT(BURIAL_PERMIT_ISSUE_DATE,4) + LEFT(BURIAL_PERMIT_ISSUE_DATE,6)),'/','') + '6666'
+					THEN REPLACE ((RIGHT(BURIAL_PERMIT_ISSUE_DATE_RV,4) + LEFT(BURIAL_PERMIT_ISSUE_DATE_RV,6)),'/','') + '6666'
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:570]: C.CURRENT_CITY_OPT_IN = 'Y' AND BURIAL_PERMIT_ISSUE_DATE IS NULL AND ISDATE(BURIAL_PERMIT_ISSUE_DATE_RV)

is there case BURIAL_PERMIT_ISSUE_DATE_RV NULL. This logic might give NULL

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 2]
```diff
@@ -288,9 +567,9 @@ END
...
 				  WHEN C.CURRENT_CITY_OPT_IN = 'Y' AND BURIAL_PERMIT_ISSUE_DATE IS NULL AND ISDATE(BURIAL_PERMIT_ISSUE_DATE_RV) = 1 
-					THEN (CONVERT(VARCHAR(30),CONVERT (DATETIME , BURIAL_PERMIT_ISSUE_DATE_RV ),112 ) + '6666')
+					THEN (CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME , BURIAL_PERMIT_ISSUE_DATE_RV )),112 ) + '6666')
 				  WHEN C.CURRENT_CITY_OPT_IN = 'Y' AND ISDATE(BURIAL_PERMIT_ISSUE_DATE) = 0
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:573]: what is BURIAL_PERMIT_ISSUE_DATE null

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 3]
```diff
@@ -300,43 +579,106 @@ END
...
-+  CASE WHEN BURIAL_PERMIT_NONE = 'Y' THEN ''
-		WHEN BURIAL_PERMIT_NONE = 'N' AND BURIAL_PERMIT_NUMBER IS NULL AND BURIAL_PERMIT_NUMBER_RV IS NULL THEN ''
-		WHEN BURIAL_PERMIT_NONE = 'N' THEN
-			(CASE WHEN C.CURRENT_CITY_OPT_IN = 'Y' THEN ISNULL(BURIAL_PERMIT_NUMBER,BURIAL_PERMIT_NUMBER_RV)
-				  ELSE ISNULL(BURIAL_PERMIT_NUMBER,'') END)
+	END						 --259		SCRAMBLED
++ '|' + CASE WHEN BURIAL_PERMIT_NONE = 'Y' THEN ''
+			 WHEN BURIAL_PERMIT_NONE = 'N' AND BURIAL_PERMIT_NUMBER IS NULL AND BURIAL_PERMIT_NUMBER_RV IS NULL THEN ''
+			 WHEN BURIAL_PERMIT_NONE = 'N' THEN
+				(CASE WHEN C.CURRENT_CITY_OPT_IN = 'Y' THEN ISNULL(BURIAL_PERMIT_NUMBER,BURIAL_PERMIT_NUMBER_RV)
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:586]: what is BURIAL_PERMIT_NUMBER_RV null in this case

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 4]
```diff
@@ -1304,56 +1768,935 @@ END
...
+			 WHEN ATTEND_PHYSICIAN_TITLE = 2 THEN 'DO'
+			 WHEN ATTEND_PHYSICIAN_TITLE = 3 THEN 'NP'
+			 WHEN ATTEND_PHYSICIAN_TITLE = 4 THEN 'PA'
+			 --WHEN ATTEND_PHYSICIAN_TITLE = 8 THEN 'Other'			--COMMENDTED OUT AS PER CODE REVIEW BECAUSE NO 'OTHER' IN DATA
+			 WHEN ATTEND_PHYSICIAN_TITLE IS NULL THEN ''
+		END		--+ ISNULL(CAST(ATTEND_PHYSICIAN_TITLE AS VARCHAR),'') --772 check att-s
++ '|'														--773
++ '|' + CASE WHEN ATTEND_PHYSICIAN_NONE = 'Y' THEN '1'
+			 WHEN ATTEND_PHYSICIAN_NONE = 'N' THEN '0'
+		END													--774
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2129]: ATTEND_PHYSICIAN_NONE  has null

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 5]
```diff
@@ -1304,56 +1768,935 @@ END
...
+			 WHEN VET1_BRANCH = 'PUBLIC HEALTH SERVICE' THEN 'Public Health Service'
+			 WHEN VET1_BRANCH IS NULL THEN ''
+		END													--791
++ '|' + CASE --WHEN VET1_DATE_ENTERED = '99/99/9999' THEN '999999996666' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+			 WHEN VET1_DATE_ENTERED IS NULL THEN ''
+			 --WHEN VET1_DATE_ENTERED LIKE 'N%A' THEN '666666666666'
+			 WHEN ISDATE(VET1_DATE_ENTERED)= 0 AND REPLACE(VET1_DATE_ENTERED,'9','') IN ('/', '') THEN '99999999' + '6666'   -- dont need toi part only keep 6666
+			 WHEN ISDATE(VET1_DATE_ENTERED)= 0 THEN  REPLACE ((RIGHT(VET1_DATE_ENTERED,4) + LEFT(VET1_DATE_ENTERED,6)),'/','') + '6666'
+			 WHEN ISDATE(VET1_DATE_ENTERED)= 1 AND LEN(VET1_DATE_ENTERED)= 4 THEN VET1_DATE_ENTERED + '9999' + '6666'
+			 WHEN ISDATE(VET1_DATE_ENTERED)= 1 THEN ISNULL(CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME, VET1_DATE_ENTERED)),112),'')  + '6666'
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2180]: data fixes have been sent right?
10/2/1958
03/4/1955
04/6/1991
10/251951

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 6]
```diff
@@ -1304,56 +1768,935 @@ END
...
+		END													--792		SCRAMBLED
++ '|' + CASE --WHEN VET1_DATE_DISCHARGED = '99/99/9999' THEN '999999999999' --SHOULD THIS BE 99999999 OR LEFT BLANK IF YES THEN RECHECK FOR DOI
+			 WHEN VET1_DATE_DISCHARGED IS NULL THEN ''
+			 WHEN VET1_DATE_DISCHARGED LIKE 'N%A' THEN '666666666666'
+			 WHEN ISDATE(VET1_DATE_DISCHARGED)= 0 AND REPLACE(VET1_DATE_DISCHARGED,'9','') IN ('/', '') THEN '99999999' + '6666'    -- Check MAc incorrect value for DOI it db
+			 WHEN ISDATE(VET1_DATE_DISCHARGED)= 0 THEN  REPLACE ((RIGHT(VET1_DATE_DISCHARGED,4) + LEFT(VET1_DATE_DISCHARGED,6)),'/','') + '6666'
+			 WHEN ISDATE(VET1_DATE_DISCHARGED)= 1 AND LEN(VET1_DATE_DISCHARGED)=4 THEN VET1_DATE_DISCHARGED + '9999' + '6666'
+			 WHEN ISDATE(VET1_DATE_DISCHARGED)= 1 THEN ISNULL(CONVERT(VARCHAR(30),DATEADD(DAY,+2,CONVERT (DATETIME, VET1_DATE_DISCHARGED)),112),'')  + '6666'
+		END													--793		SCRAMBLED
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2189]: in correct dates were sent right 
02/101971
1956
07/311951
9/14/1955
09/301967
09/2/1954
N/A

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 7]
```diff
@@ -1304,56 +1768,935 @@ END
...
++ '|'					--850
++ '|'					--851
++ '|'					--852
++ '|'					--853
++ '|'					--854
++ '|'					--855
++ '|'					--856
++ '|' + CASE WHEN FL_COD_PENDING = 'Y' THEN '1'
+			 WHEN FL_COD_PENDING = 'N' THEN '0'
+		END				--857
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2337]: it does not have null but it's good to add NULL case

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 8]
```diff
@@ -1304,56 +1768,935 @@ END
...
++ '|' --918
++ '|' --919
++ '|' --920
++ '|' --921
++ '|' --922
++ '|' --923
++ '|' --924
++ '|' + CASE WHEN FL_SIGNED_BEHALF_CERT = 'Y' THEN '1'
+		     WHEN FL_SIGNED_BEHALF_CERT = 'N' THEN '0'
+		END		--925
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2452]: Handle NULL Case

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 9]
```diff
@@ -1304,56 +1768,935 @@ END
...
+			 WHEN ST_REGIS_NAME=4 THEN 'MICHAEL CRUTCHER'
+			 WHEN ST_REGIS_NAME=5 THEN 'STANLEY E. NYBERG'
+			 WHEN ST_REGIS_NAME=6 THEN 'GERALD F. O''KEEFE'
+			 WHEN ST_REGIS_NAME=7 THEN 'ANTONIO C. SOUSA'
+			 WHEN ST_REGIS_NAME=8 THEN 'DEAN DIMARTINO'
+			 WHEN ST_REGIS_NAME=9 THEN 'ELAINE B. TRUDEAU'
+			 WHEN ST_REGIS_NAME=10 THEN 'DANIEL J. FRIEDMAN'
+			 WHEN ST_REGIS_NAME=11 THEN 'KARIN A. BARRETT'
+			 ELSE '' 
+		END											--984	PART OF SEPT RELEASE
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2621]: You need to revisit list before during prod

**Review by almabekwoldesenbet — COMMENTED** [review]
required change

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 10]
```diff
@@ -165,36 +385,42 @@ END
 			 WHEN CERT_DESIG = 'PHYSICIAN ASSISTANT IN ATTENDANCE AT TIME OF DEATH' AND FL_PRONOUNCEMENT_EXISTS = 'N' THEN 'Pronouncing and Certifying Practitioner'
 			 WHEN CERT_DESIG = 'PHYSICIAN ASSISTANT IN CHARGE OF PATIENT''S CARE' AND FL_PRONOUNCEMENT_EXISTS = 'Y' THEN 'Certifying Practitioner'
 			 WHEN CERT_DESIG = 'PHYSICIAN ASSISTANT IN CHARGE OF PATIENT''S CARE' AND FL_PRONOUNCEMENT_EXISTS = 'N' THEN 'Pronouncing and Certifying Practitioner'
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:387]: NULL case is missing

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 11]
```diff
@@ -1304,56 +1768,935 @@ END
...
++ '|' + ISNULL(CAST(ME_CASE_YEAR AS VARCHAR),'')			--759 
++ '|'									--760
++ '|' + ISNULL(CT.Mapping_Current_PhysicanFirstName,'')		--761
++ '|' + ISNULL(CT.Mapping_Current_PhysicanLastName,'')		--762
++ '|' + ISNULL(CT.Mapping_Current_PhysicanMiddleName,'')	--763
++ '|' + ISNULL(CT.Mapping_Current_PhysicanSuffix,'')		--764
++ '|' + ISNULL(CT.Mapping_Current_PhysicanTitle,'')			--765
++ '|'														--766	--Title Other in not in VIP, check with Bezawit
++ '|' + CASE WHEN FL_CERT_ATTEND_PHYSICIAN ='Y' THEN '0'
+			 ELSE '1'
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2113]: we need to send blank

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql [hunk 12]
```diff
@@ -1304,56 +1768,935 @@ END
...
+			 WHEN UNITD = 'HRS.' THEN 'Hours'
+			 WHEN UNITD = 'DAYS' THEN 'Days'
+			 WHEN UNITD = 'WKS.' THEN 'Weeks'		 
+			 WHEN UNITD = 'MOS.' THEN 'Months'
+			 WHEN UNITD = 'YRS.' THEN 'Years'
+		END
+		--+ ISNULL(UNITD,'')--861
++ '|' + CASE WHEN FL_PRONOUNCEMENT_EXISTS = 'Y' THEN 'Yes'
+			 WHEN FL_PRONOUNCEMENT_EXISTS = 'N' THEN 'No'
+			 ELSE ''
```
Comment (almabekwoldesenbet) [inline RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2380]: should this be a data fixes

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 1 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:570]: no records that finds this condition

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 2 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:573]: we don't have cases where CURRENT_CITY_OPT_IN = 'Y' 
AND BURIAL_PERMIT_ISSUE_DATE IS NULL 
AND BURIAL_PERMIT_ISSUE_DATE_RV is null

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 3 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:586]: we don't have any case where
C.CURRENT_CITY_OPT_IN = 'Y' 
AND BURIAL_PERMIT_NUMBER IS NULL 
AND BURIAL_PERMIT_NUMBER_RV is null

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 4 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2129]: FIXED

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 5 above)
Comment (sailendrasingh1) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:2180]: it will be data fix and we will revisit this again.

**Review by almabekwoldesenbet — COMMENTED** [review]
looks good

File: RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql (hunk 10 above)
Comment (almabekwoldesenbet) [reply RVRS_Staging/Database/Procedure/FOR_MASTER_GENERIC_DATA_SCRAMBLING.sql:390]: No need to add NULL for registered records

**Comment by almabekwoldesenbet** [comment]
2021 pre-production code - First data migration

**Pull request merged**